### DISCUSSION TITLE
Automatic charge and discharge feature

### DISCUSSION THREAD

**Discussion by muhdshafiqsofian** [mhaeuser/Battery-Toolkit#30] [Ideas]
Hi,

It will be great if we can have automatic charge and discharge feature. This is somehow will give protection to the battery.

I have Lenovo laptop that have this feature and margin been set is about 5% as shown in picture.

![WhatsApp Image 2024-07-30 at 13 34 50](https://github.com/user-attachments/assets/e4c025bf-95c5-47cd-970a-97b72f15e105)

Thanks in advance!

**Comment by mhaeuser** [comment]
This will not conserve the battery, this will wear it to pieces. It indeed is expected you use the device off-plug sometimes. If you don’t, you can use Disable Power Adapter every now and then. Maybe I could add a feature to automate this some day, but you’d want a proper cycle between 20 % and 80 % and only sometimes, not constant changing and discharging between 75 % and 80 %.
